### PR TITLE
Add user activation, metadata, routes, and soft delete logic; fix response path and server errors

### PR DESCRIPTION
**Pull request by bellaabdelouahab**
related issues #54 #22 

here effected tasks in general 

- [x] Add activation requirement for new signup users (optional via env).
- [X] Add metadata fields: deleted, updatedBy, createdBy, deletedBy, deletedAt.
- [X] Implement routes: activate, updateMyPassword, forgotPassword.
- [X] Add soft delete logic (requires extensive testing).
- [X] Fix server errors displayed on production environment.

@yassineboujrada @Khalid1G @muttaqin1 
please provide some feedbacks

### CODE REVIEW

File: backend-app/controllers/auth_controller.js [hunk 1]
```diff
@@ -108,6 +116,141 @@ exports.signup = async (req, res, next) => {
...
+};
+
+
+exports.updatePassword = async (req, res, next) => {
+  try {
+    const { email, resetKey, password } = req.body;
+
+    const user = await User.findOne({ email }).select("+password");
+
+    if (!resetKey) {
```
Comment (muttaqin1) [inline backend-app/controllers/auth_controller.js:189]: This status code can be 400

File: backend-app/controllers/auth_controller.js [hunk 2]
```diff
@@ -108,6 +116,141 @@ exports.signup = async (req, res, next) => {
...
+      return next(new AppError(404, "fail", "Please provide reset key"));
+    }
+
+    if (resetKey !== user.resetKey) {
```
Comment (muttaqin1) [inline backend-app/controllers/auth_controller.js:193]: This status code can be 403

File: backend-app/controllers/auth_controller.js [hunk 3]
```diff
@@ -108,6 +116,141 @@ exports.signup = async (req, res, next) => {
...
+    next(err);
+  }
+};
+
+
+exports.forgotPassword = async (req, res, next) => {
+  try {
+    const { email } = req.body;
+
+    if (!email) {
```
Comment (muttaqin1) [inline backend-app/controllers/auth_controller.js:219]: This status code can be 400

**Review by muttaqin1 — CHANGES_REQUESTED** [review]

**Review by muttaqin1 — APPROVED** [review]
All seems fine.

**Review by Khalid1G — APPROVED** [review]

File: backend-app/controllers/auth_controller.js (hunk 3 above)
Comment (bellaabdelouahab) [reply backend-app/controllers/auth_controller.js:219]: sure

File: backend-app/app.js [hunk 4]
```diff
@@ -72,7 +72,7 @@ app.get('/', (req, res) => {
 
 // handle undefined Routes
 app.use('*', (req, res, next) => {
-  const err = new AppError(404, 'fail', 'Route Not Found', req.path);
+  const err = new AppError(404, 'fail', 'Route Not Found', req.originalUrl);
```
[the remark this answers is not in the document]
Comment (bellaabdelouahab) [reply backend-app/app.js:75]: sure

File: backend-app/constants/meta_data.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            default: false
+        },
+        UpdatedBy: {
+            type: String,
+        },
+        createdBy: {
+            type: String,
+            default: 'System'
+        },
+        deletedBy: {
```
[the remark this answers is not in the document]
Comment (bellaabdelouahab) [reply backend-app/constants/meta_data.js:16]: and how we should handle system ?

File: backend-app/middlewares/global_error_handler.js [hunk 6]
```diff
@@ -27,16 +28,22 @@ module.exports = (err, req, res, next) => {
...
   err.message = err.message || httpStatus.getStatusText(err.statusCode);
-
+  console.log(CURRENT_ENV);
   res.status(err.statusCode).json({
     status: err.statusCode,
     title: httpStatus.getStatusText(err.statusCode),
     details: {
       ...(err.path && { path: err.path }),
-      description: err.message,
+      description:
```
[the remark this answers is not in the document]
Comment (bellaabdelouahab) [reply backend-app/middlewares/global_error_handler.js:37]: what do you mean this is the most optimized way, I think it needs just some comments
give an example of what is hard to read

File: backend-app/constants/meta_data.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Khalid1G) [reply backend-app/constants/meta_data.js:16]: create account system and put that ID

File: backend-app/middlewares/global_error_handler.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (bellaabdelouahab) [reply backend-app/middlewares/global_error_handler.js:37]: sure i ll fix it in next pr

**Review by muttaqin1 — APPROVED** [review]
great👌

File: backend-app/constants/meta_data.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (bellaabdelouahab) [reply backend-app/constants/meta_data.js:16]: no the right way

File: backend-app/controllers/auth_controller.js (hunk 2 above)
Comment (bellaabdelouahab) [reply backend-app/controllers/auth_controller.js:193]: 👌

**Comment by bellaabdelouahab** [comment]
@Khalid1G

**Pull request merged**